packages/vue-components/components/profile/update-email/update-email.vue

```diff
@@ -0,0 +1,98 @@
+<template>
+  <generic-form
+    :fields
+    :buttons="{
+      submit: true,
+      reset: false,
+      cancel: false,
+    }"
+    :labels="{
+      submit: t('locokit.components.updateEmailForm.submit'),
+    }"
+    :loading="loading"
+    :message="message"
+    @submit="onSubmit"
+  >
+    <template #top>
+      <div class="mb-4">
+        <p>{{ $t('locokit.components.updateEmailForm.currentEmail') }}</p>
+        <p class="font-bold">{{ user?.email }}</p>
+      </div>
+    </template>
+  </generic-form>
+</template>
+
+<script setup lang="ts">
+import { computed } from 'vue'
+import { useI18n } from 'vue-i18n'
+import { FIELD_COMPONENT, FIELD_TYPE, LocoKitFormField, LocoKitMessage } from '@locokit/definitions'
+import GenericForm from '@/components/commons/generic-form/generic-form.vue'
+
+const { t } = useI18n()
+
+const emit = defineEmits<{
+  /**
+   * Emitted when the submit button has been clicked
+   * and the form has been successfully validated.
+   */
+  (
+    e: 'submit',
+    form: {
+      id: string
+      newEmail: string
+      password: string
+    },
+  ): void
+}>()
+
+const props = withDefaults(
+  defineProps<{
+    /** The user object concerned by the update. */
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    user: any
```

Comment:
Maybe we should define the `LocoKitUser` type ? 
Or at least the shape, with an `id` as you use later `props.user.id`.

Reply:
You remember me that I wanted to talk to you about the necessity of passing a user object to this component. We only use it to include its `id` in the emitted data. I think we should remove it in this case. The `update-password.vue` component, which is similar, does not expect such an object.

What do you think?

> Maybe we should define the LocoKitUser type ?

Yes, I think so. I will add this new type.

Reply:
As discussed together:

- `update-email.vue` component revised to no longer expect a complete user object
- new type `LocoKitUser` created and used in `update-general.vue` component